### PR TITLE
Added Clinical Trials tool. Changes made in docker-compose.yml to work with GPU. Streaming Final Answer of the Agent

### PR DESCRIPTION
**Pull request by felipegm0911**

### CODE REVIEW

File: backend/python-langchain/main.py [hunk 1]
```diff
@@ -34,12 +34,18 @@
 from langchain import LLMChain
 from typing import List, Union
 from langchain.schema import AgentAction, AgentFinish, OutputParserException
-import re
 from langchain.prompts import PromptTemplate
 from src.pupmed import PubMedRetriever
 from langchain.memory import ConversationBufferMemory, ReadOnlySharedMemory
+from langchain.callbacks.base import BaseCallbackHandler
+from langchain.embeddings import GPT4AllEmbeddings
```
Comment (bowheadadmin) [inline backend/python-langchain/main.py:41]: @felipegm0911 if this is not needed, please remove it

File: backend/python-langchain/main.py [hunk 2]
```diff
@@ -49,18 +55,11 @@
 api_url = getenv('LANGCHAIN_UPLOAD_ENDPOINT')
 delete_url = getenv('LANGCHAIN_DELETE_ENDPOINT')
 cors_domains = getenv('CORS_DOMAINS').split(',')
-cookie_domain = getenv('COOKIE_DOMAIN')
 app = Flask(__name__)
 app.secret_key = getenv('SECRET_KEY', '')
 app.config['SESSION_TYPE'] = 'filesystem'
-app.config['SESSION_COOKIE_SAMESITE'] = 'None'
```
Comment (bowheadadmin) [inline backend/python-langchain/main.py:56]: @felipegm0911 please return back all these cookie configurations, these are needed to handle the front end connection

File: backend/python-langchain/main.py [hunk 3]
```diff
@@ -49,18 +55,11 @@
...
-app.config['SESSION_COOKIE_SECURE'] = True
-app.config['SESSION_COOKIE_DOMAIN'] = cookie_domain
-app.config['REMEMBER_COOKIE_SAMESITE'] = 'None'
-app.config['REMEMBER_COOKIE_SECURE'] = True
-app.config['REMEMBER_COOKIE_DOMAIN'] = cookie_domain
-
-CORS(app, supports_credentials=True, origins=cors_domains)
```
Comment (bowheadadmin) [inline backend/python-langchain/main.py:63]: @felipegm0911 please return the origins parameter, this is used int he staging environment

File: backend/python-langchain/main.py [hunk 4]
```diff
@@ -236,6 +246,24 @@ def send_message():
         input_key="question"
     )
 
+    # ************ MEDICAL TRIALS VECTOR ************
+    instructor_embeddings = HuggingFaceInstructEmbeddings(model_name = 'hkunlp/instructor-base', model_kwargs = {"device": "cuda"})
```
Comment (bowheadadmin) [inline backend/python-langchain/main.py:250]: @felipegm0911 do we need GPU?

**Review by bowheadadmin — CHANGES_REQUESTED** [review]

File: backend/python-langchain/main.py (hunk 4 above)
Comment (felipegm0911) [reply backend/python-langchain/main.py:250]: yes, the vector was created with GPU. I could check if we can use the same vector without GPU, but if it works, the response would be slower

**Review by bowheadadmin — APPROVED** [review]

**Pull request merged**